python/tests/intervals_tests.py

```diff
@@ -0,0 +1,344 @@
+from tempo.intervals import *
+from tests.tsdf_tests import SparkTest
+from pyspark.sql.utils import AnalysisException
+import pyspark.sql.functions as f
+
+
+class IntervalsDFTests(SparkTest):
+    union_tests_dict_input = [
+        {
+            "start_ts": "2020-08-01 00:00:09",
+            "end_ts": "2020-08-01 00:00:14",
+            "series_1": "v1",
+            "metric_1": 5,
+            "metric_2": None,
+        },
+        {
+            "start_ts": "2020-08-01 00:00:09",
+            "end_ts": "2020-08-01 00:00:11",
+            "series_1": "v1",
+            "metric_1": None,
+            "metric_2": 0,
+        },
+        {
+            "start_ts": "2020-08-01 00:00:09",
+            "end_ts": "2020-08-01 00:00:12",
+            "series_1": "v1",
+            "metric_1": None,
+            "metric_2": 4,
+        },
+        {
+            "start_ts": "2020-08-01 00:00:09",
+            "end_ts": "2020-08-01 00:00:14",
+            "series_1": "v1",
+            "metric_1": 5,
+            "metric_2": None,
+        },
+        {
+            "start_ts": "2020-08-01 00:00:09",
+            "end_ts": "2020-08-01 00:00:11",
+            "series_1": "v1",
+            "metric_1": None,
+            "metric_2": 0,
+        },
+        {
+            "start_ts": "2020-08-01 00:00:09",
+            "end_ts": "2020-08-01 00:00:12",
+            "series_1": "v1",
+            "metric_1": None,
+            "metric_2": 4,
+        },
+    ]
+
+    def test_init_series_str(self):
+        df_input = self.get_data_as_sdf("input")
+
+        self.assertRaises(
+            ValueError,
+            IntervalsDF,
+            df_input,
+            "start_ts",
+            "end_ts",
+            "series_1",
+        )
+
+    def test_init_series_tuple(self):
+        df_input = self.get_data_as_sdf("input")
+
+        self.assertRaises(
+            ValueError,
+            IntervalsDF,
+            df_input,
+            "start_ts",
+            "end_ts",
+            ("series_1",),
+        )
+
+    def test_init_series_list(self):
```

Comment:
@R7L208 , fantastic work here. Can we add a representative intervals example in our examples folder? Perhaps with DLT? I'm happy to help document.

Reply:
Yes, let's definitely do that. I'd like to cut a new release ASAP so that users can start using this. Ok with you if we include the docs in a later update?

Reply:
That's fine.